File: src/main/java/com/wikia/tibia/repositories/BuildingRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.BuildingGateway;
import com.wikia.tibia.objects.Building;

public class BuildingRepository extends WikiObjectRepository {

    public BuildingRepository() {
        super(Building.class, new BuildingGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/BuildingRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Building

class BuildingRepository : WikiObjectRepository<Building>(
  wikiObjectClass = Building::class.java,
  wikiObjectGateway = WikiObjectGateway(Contract.BUILDINGS)
)
